internal/plugin/server.go

```diff
@@ -69,41 +66,25 @@ type NvidiaDevicePlugin struct {
 	mpsHostRoot mps.Root
 }
 
-// NewNvidiaDevicePlugin returns an initialized NvidiaDevicePlugin
-func NewNvidiaDevicePlugin(config *spec.Config, kubeletSocket string, resourceManager rm.ResourceManager, cdiHandler cdi.Interface) (*NvidiaDevicePlugin, error) {
-	_, name := resourceManager.Resource().Split()
-
-	deviceListStrategies, _ := spec.NewDeviceListStrategies(*config.Flags.Plugin.DeviceListStrategy)
-
-	pluginName := "nvidia-" + name
-	pluginPath := filepath.Join(pluginapi.DevicePluginPath, pluginName)
-
-	var mpsDaemon *mps.Daemon
-	var mpsHostRoot mps.Root
-	if config.Sharing.SharingStrategy() == spec.SharingStrategyMPS {
-		// TODO: It might make sense to pull this logic into a resource manager.
-		for _, device := range resourceManager.Devices() {
-			if device.IsMigDevice() {
-				return nil, errors.New("sharing using MPS is not supported for MIG devices")
-			}
-		}
-		mpsDaemon = mps.NewDaemon(resourceManager, mps.ContainerRoot)
-		mpsHostRoot = mps.Root(*config.Flags.CommandLineFlags.MpsRoot)
+// devicePluginForResource creates a device plugin for the specified resource.
+func (o *options) devicePluginForResource(resourceManager rm.ResourceManager) (Interface, error) {
+	mpsDaemon, mpsHostRoot, err := o.getMPSDaemon(resourceManager)
+	if err != nil {
+		return nil, err
 	}
 
-	plugin := NvidiaDevicePlugin{
+	plugin := nvidiaDevicePlugin{
 		rm:                   resourceManager,
-		config:               config,
-		deviceListEnvVar:     "NVIDIA_VISIBLE_DEVICES",
-		deviceListStrategies: deviceListStrategies,
-		socket:               pluginPath + ".sock",
-		cdiHandler:           cdiHandler,
-		cdiAnnotationPrefix:  *config.Flags.Plugin.CDIAnnotationPrefix,
```

Comment:
We originally included the `deviceListEnvVar` here in case there was a *different* envvar that we needed to use to pass the device info back to the runtime. In practice this was never necessary, so it should be fine to remove this level of configurability.

Reply:
I would rather clean this up here and readd it if we see that we need it. This struct was getting quite big already and I wanted to trim it down.

We also don't allow this level of configurability for other envvars.